auth/design.md

```diff
@@ -0,0 +1,68 @@
+﻿
+# etcd v3 authentication design
+
+## Why not using the existing v2 auth mechanism?

```

Comment:
`## Why not reuse the v2 auth system?` ?

Also, please add a newline between headers and text:
```md
## Whatever

Some text
```